app/controllers/ops_controller/settings/rhn.rb

```diff
@@ -256,11 +258,12 @@ def rhn_validate
     rhn_load_session
     rhn_fire_available_organizations do |organizations|
       if 'rhn_satellite6' == @edit[:new][:register_to]
+        # Hash of display names to key names
         @edit[:organizations] = organizations
         if @edit[:organizations].length == 1
-          @edit[:new][:customer_org] = @edit[:organizations].first
+          @edit[:new][:customer_org_display], @edit[:new][:customer_org] = @edit[:organizations].first
         else
-          @edit[:new][:customer_org] = nil unless @edit[:organizations].include?(@edit[:new][:customer_org])
+          @edit[:new][:customer_org] = nil unless @edit[:organizations].key?(@edit[:new][:customer_org_display])
```

Comment:
It gets pulled from the `@edit[:organizations]` hash either way on save [here](https://github.com/ManageIQ/manageiq/pull/7893/files#diff-ec61b7de06c989da9295e407e8943867R130), so I guess setting it in the `if` block is useless, but might look better than ignoring that part of the `@edit[:organizations].first` array